src/becca/becca-interface.ts

```diff
@@ -1,32 +1,51 @@
-"use strict";
-
-const sql = require('../services/sql.js');
-const NoteSet = require('../services/search/note_set.js');
-const NotFoundError = require('../errors/not_found_error.js');
+import sql = require('../services/sql');
+import NoteSet = require('../services/search/note_set');
+import NotFoundError = require('../errors/not_found_error');
+import BOption = require('./entities/boption');
+import BNote = require('./entities/bnote');
+import BEtapiToken = require('./entities/betapi_token');
+import BAttribute = require('./entities/battribute');
+import BBranch = require('./entities/bbranch');
+import BRevision = require('./entities/brevision');
+import BAttachment = require('./entities/battachment');
+import { AttachmentRow, RevisionRow } from './entities/rows';
```

Comment:
@ProfDoof , it's a nice proposal. I guess the main reason I did not do ESM conversion yet is not just the async stuff, but also the amount of changes that I had to do.

In order for `import` to work, I also need to replace `module.exports` with `export {`. And I wanted to use `export` at method level instead of a big one at the end of modules. This created too many changes and I wanted us to focus first on actually converting the code base.

It should be much easier to review the module conversion in a separate PR, since we can triage the modifications more easily.

What do you think?

Reply:
I think that you're probably right. I will admit to a proclivity for accidentally scope-creeping myself.